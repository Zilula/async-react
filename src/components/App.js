import React, { Component, Fragment } from 'react';
import Header from './header/Header';
import Characters from './characters/Characters';

export default class App extends Component {
    state = {
        currentPage: 1,
        totalPages: null,
    }


    decrementPage = () => {
        this.setState({ currentPage: this.state.currentPage - 1 });
    }
    incrementPage = () => {
        this.setState({ currentPage: this.state.currentPage + 1 });
    }
    // updateTotalPages = () => {
    //     this
    // }


    render() {
        const { currentPage, totalPages } = this.state;
        console.log(currentPage);
        // const pageNum = currentPage.toString();
        return (
            <Fragment>
                <Header/>
                <p>CURRENT PAGE {currentPage}</p>
                <p>TotalPages {totalPages}</p>
                <button onClick={this.incrementPage}>NEXT PAGE</button>
                <button onClick={this.decrementPage}>PREVIOUS PAGE</button>
                <Characters
                    currentPage={currentPage}/>
            </Fragment>
        );
    }
}
