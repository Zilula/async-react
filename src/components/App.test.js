import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';


describe('testing things', () => {


    it('renders the component correctly', () => {
        const tree = renderer.create(
            <App/>
        ).toJSON();
        expect(tree).toMatchSnapshot();

    });
});
