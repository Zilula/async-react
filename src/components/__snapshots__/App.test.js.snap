// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing things renders the component correctly 1`] = `
Array [
  <h1
    className="header"
  >
    HEADER Component
  </h1>,
  <ul />,
]
`;
